### DISCUSSION TITLE
Social Authentication W/ NextJS API Endpoint

### DISCUSSION THREAD

**Discussion by tommyboylab** [strapi/strapi#6212] [Legacy Topics]
I've been trying to find a tutorial on how to properly handle the OAuth callback from my BE which is on a subdomain of my site (Strapi) to my FE NextJS server. 

So far I've managed to successfully setup a component that redirects the user to the provider, retrieve the access token as a query parameter, and by redirecting to the Strapi callback endpoint, retrieve the JSON object with my data.

I'm now looking to how I can redirect the user back to my FE application, and store that data in a cookie for use. So far I've been unable to understand the proper way to handle this. If someone could please help me understand how I can achieve this I'd greatly appreciate it.

I was thinking it might be possible to use a NextJS api endpoint to which the Strapi callback is directed, from there the query parameters are retrieved, added to the url to be sent to the Strapi API endpoint where they are checked for validity, and if valid, the data is stored in a cookie.

If someone could help me with an example of the process for the FE, I'd be a great help.

Link to the StackOverflow question I asked a few days ago:
https://stackoverflow.com/questions/61725122/strapi-social-auth-w-nextjs

**Comment by lauriejim** [comment]
Hello! Here is a video that explains the process.
I think it could help you.
https://youtu.be/s_vQqQ4tpHs
